GAE/src/com/gallatinsystems/survey/dao/SurveyUtils.java

```diff
@@ -111,14 +113,14 @@ public static void copySurvey(Long copiedSurveyId, Long originalSurveyId) {
 
         log.log(Level.INFO, "Copying " + qgList.size() + " `QuestionGroup`");
 
-        Map<Long, QuestionGroup> sourceToCopiedGroupMap = shallowCopyQuestionGroups(copiedSurveyId, qgList);
+        List<Pair<Long, QuestionGroup>> sourceToCopiedGroupMap = shallowCopyQuestionGroups(copiedSurveyId, qgList);
 
         // batch save question groups
-        new QuestionGroupDao().save(sourceToCopiedGroupMap.values());
+        new QuestionGroupDao().save(sourceToCopiedGroupMap.stream().map(pair -> pair.right).collect(Collectors.toList()));
```

Comment:
this complex one can be done with `new ArrayList(sourceToCopiedGroupMap.values())` ? not sure the value of stream to just get the values

Reply:
yes because this is no longer a map, I renamed the var to reflect that